pilot/pkg/model/policyattachment.go

```diff
@@ -143,9 +136,20 @@ func (p WorkloadPolicyMatcher) ShouldAttachPolicy(kind config.GroupVersionKind,
 		if p.IsWaypoint &&
 			matchesGroupKind(targetRef, gvk.ServiceEntry) &&
 			target == p.Service &&
+			policyName.Namespace == p.ServiceNamespace &&
 			p.ServiceRegistry == provider.External {
 			return true
 		}
+
+		// Namespace does not match
+		if !(targetRef.GetNamespace() == "" || targetRef.GetNamespace() == p.WorkloadNamespace) {
+			continue
```

Comment:
could it be useful to log.Debugf that they don't match? something like:
```go
log.Debugf("Skipping targetRef %s in namespace %s for workload in namespace %s", targetRef.GetName(), targetRef.GetNamespace(), p.WorkloadNamespace)
    continue
}
```

Or is this a fairly common occurrence that doesn't necessarily indicate any issues in this path?

Reply:
Same comment, this is going to apply to every policy, so its totally expected many will not match.

Maybe the type of thing https://github.com/istio/istio/issues/43030 would be decent for